homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -652,6 +652,17 @@ async def async_select_source(self, source: str) -> None:
 
     async def async_select_sound_mode(self, sound_mode: str) -> None:
         """Select a sound mode."""
+        # Ensure only known sound modes known by the integration can be activated.
+        if sound_mode not in self._sound_modes:
+            raise ServiceValidationError(
```

Comment:
Please add a test which exercises this path

Reply:
Ah sorry. A test has been added now.